.github/workflows/CI-CD.yml

```diff
@@ -23,10 +23,20 @@ jobs:
     name: "Build and Test"
     strategy:
       matrix:
-        os: [windows-latest, ubuntu-latest]
+        os: [windows-latest, ubuntu-latest, macos-12]
```

Comment:
Does this not work on `macos-13` and `macos-14`, or is it just not tested on them?

Reply:
For now I'm going for consistency and testability, and I only have access to a functioning macOS 12 VM myself. A bigger concern seems to be Intel silicon vs Apple silicon as the unit tests are crashing on Apple silicon based on some feedback I've heard. I'm trying to get us to at least a point where we can show that the code is demonstrably working for some macOS computers. Further changes will probably come later to fix more issues we can't easily automate testing for at the moment.